src/flow/src/adapter/table_source.rs

```diff
@@ -123,25 +122,34 @@ impl TableSource {
         ];
 
         let raw_schema = table_info_value.table_info.meta.schema;
-        let column_types = raw_schema
+        let (col_names, column_types): (Vec<_>, Vec<_>) = raw_schema
             .column_schemas
+            .clone()
             .into_iter()
-            .map(|col| ColumnType {
-                nullable: col.is_nullable(),
-                scalar_type: col.data_type,
+            .map(|col| {
+                (
+                    col.name.clone(),
```

Comment:
I guess the clone can be eliminated
```suggestion
                    col.name,
```